grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/resilience/ResilienceCircuitBreakerProvider.java

```diff
@@ -0,0 +1,59 @@
+package org.hypertrace.circuitbreaker.grpcutils.resilience;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerConfig;
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+
+/** Utility class to provide Resilience4j CircuitBreaker */
+@Slf4j
+class ResilienceCircuitBreakerProvider {
+
+  private final CircuitBreakerRegistry circuitBreakerRegistry;
+  private final Map<String, CircuitBreakerConfig> circuitBreakerConfigMap;
+  private final Map<String, CircuitBreaker> circuitBreakerCache = new ConcurrentHashMap<>();
+
+  public ResilienceCircuitBreakerProvider(
+      CircuitBreakerRegistry circuitBreakerRegistry,
+      Map<String, CircuitBreakerConfig> circuitBreakerConfigMap) {
+    this.circuitBreakerRegistry = circuitBreakerRegistry;
+    this.circuitBreakerConfigMap = circuitBreakerConfigMap;
+  }
+
+  public CircuitBreaker getCircuitBreaker(String circuitBreakerKey) {
```

Comment:
This should return optional. We want to find the one that matches the key, if it exists; else the default, if it exists. Else the call should not go through a circuit breaker.

Reply:
We are maintaining a cache here, if circuitBreaker key exists it is not going through circuit breaker

Reply:
Not able to understand here, i have setup meeting to discuss and close this.